x-pack/plugins/enterprise_search/server/lib/connectors/add_connector.ts

```diff
@@ -67,8 +74,23 @@ export const addConnector = async (
     index_name: string;
     is_native: boolean;
     language: string | null;
-  }
+  },
+  createDocumentIndex: boolean = true,
```

Comment:
suggestion: It's best to avoid default values for function inputs. It hides functionality from the invoker, creating a lot of potential for unintentional behavior down the line.

Reply:
That's an interesting perspective. I've always liked to grow APIs with default values, to avoid having to refactor existing usages, and polluting old areas of the code with references to newer concerns. But I can definitely remove the default, and make the usages explicit elsewhere.

Reply:
I think it's a valid approach in codebases where it's hard to refactor (either because usages are difficult to find, usually in dynamically typed languages, or code is so entangled that it's difficult to alter). But this is a very new part of the codebase and adjusting the usage elsewhere is very easy, plus static typing means there is no chance you'll forget a usage somewhere.